src/ray/common/test/ray_syncer_test.cc

```diff
@@ -306,14 +306,10 @@ struct SyncerServerTest {
     auto start = steady_clock::now();
 
     while (duration_cast<seconds>(steady_clock::now() - start).count() <= time_s) {
-      std::promise<bool> p;
-      auto f = p.get_future();
-      io_context.post([&p, predicate]() mutable { p.set_value(predicate()); }, "TEST");
-      if (f.get()) {
+      if (predicate()) {
         return true;
-      } else {
-        std::this_thread::sleep_for(1s);
```

Comment:
local grpc goes through loopback device, at worst is no worse than a fs access, which is 10s of microseconds magnitude; sleep for a second is a waste of testing time.